app/src/main/java/org/mozilla/fenix/library/bookmarks/edit/EditBookmarkFragment.kt

```diff
@@ -110,7 +109,13 @@ class EditBookmarkFragment : Fragment(), CoroutineScope {
     }
 
     override fun onPause() {
-        updateBookmarkNode(Pair(bookmark_name_edit.text, bookmark_url_edit.text))
+        try {
+            launch {
+                updateBookmarkNode(Pair(bookmark_name_edit.text.toString(), bookmark_url_edit.text.toString()))
+            }
+        } catch (e: UrlParseFailed) {
```

Comment:
Can you catch exceptions in coroutines like this?

Reply:
Oh, looks like you're doing this check inside `updateBookmarkNode`. Maybe this was unintentional?

Reply:
Oh, that happened because I was originally doing it in both cases instead of doing once in a shared suspending function. Fixing.